### PR TITLE
Introduced DataType enum

### PR DESCRIPTION
**Pull request by tlangs**
### Description

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/ubam/UbamMetadata.scala [hunk 1]
```diff
@@ -14,7 +15,7 @@ case class UbamMetadata(
   analysisType: Option[Symbol] = None,
   baitIntervals: Option[Symbol] = None,
   baitSet: Option[Symbol] = None,
-  dataType: Option[Symbol] = None,
+  dataType: Option[DataType] = None,
```
Comment (danxmoran) [inline clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/ubam/UbamMetadata.scala:18]: If you want a 2-for-1, there's a ticket to add DataType to ubam query inputs, too.

**Review by danxmoran — APPROVED** [review]

File: clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/ubam/UbamMetadata.scala (hunk 1 above)
Comment (tlangs) [reply clio-transfer-model/src/main/scala/org/broadinstitute/clio/transfer/model/ubam/UbamMetadata.scala:18]: I'll do it.

**Review by pshapiro4broad — APPROVED** [review]
Looks OK. Have we verified that all documents in prod are either `WGS` or `Exome`?

**Comment by tlangs** [comment]
We haven't verified yet, but since everything in Zamboni is controlled ontology, I don't think we'll run into issues. We should query elasticsearch before we merge this just in case though.

**Comment by danxmoran** [comment]
Looks like some other stuff has snuck in:
```
{"took":40,"timed_out":false,"_shards":{"total":5,"successful":5,"failed":0},"hits":{"total":1563241,"max_score":0.0,"hits":[]},"aggregations":{"dts":{"doc_count_error_upper_bound":0,"sum_other_doc_count":0,"buckets":[
  {"key":"WGS","doc_count":1561565},
  {"key":"RNA","doc_count":612},
  {"key":"Genome","doc_count":207}
]}}}
```

**Comment by tlangs** [comment]
Should we add those to DataType?

**Comment by danxmoran** [comment]
I'd ask KT. The alternative is to ~delete all the non-WGS records from prod, or to~ not make this an enum.

**Comment by danxmoran** [comment]
Er, nevermind about deleting the non-WGS records. To do that we'd also have to track down all the non-WGS upsert records in GCS, which would be a nightmare.

**Review by als364 — APPROVED** [review]

File: clio-util/src/main/scala/org/broadinstitute/clio/util/model/DataType.scala [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+sealed trait DataType extends EnumEntry
+
+object DataType extends Enum[DataType] {
+  override val values: IndexedSeq[DataType] = findValues
+
+  case object WGS extends DataType
+  case object Exome extends DataType
+  case object RNA extends DataType
+  case object Genome extends DataType
```
Comment (danxmoran) [inline clio-util/src/main/scala/org/broadinstitute/clio/util/model/DataType.scala]: From slack: it shouldn't cause server failures to limit these to just WGS and Exome, so let's do it and prevent any more RNA/Genome entries from getting loaded in.

Comment (danxmoran) [reply clio-util/src/main/scala/org/broadinstitute/clio/util/model/DataType.scala]: Before we merge this we should upsert some RNA records to dev, so we can test querying them post-merge.

**Pull request merged**